### PR TITLE
Fix manufacturer data lost issue (see #267)

### PR DESCRIPTION
**Pull request by jcaude**
This is the patch for issue #267, at least for the manufacturer data lost. I have added some drop() after locks, I not sure about this one, I'm used to release locks as soon as they became unnecessary.

### CODE REVIEW

File: src/winrtble/peripheral.rs [hunk 1]
```diff
@@ -143,26 +143,30 @@ impl Peripheral {
 
                 let mut local_name_guard = self.shared.local_name.write().unwrap();
                 *local_name_guard = Some(name.to_string());
+                drop(local_name_guard);
```
Comment (qwandor) [inline src/winrtble/peripheral.rs:146]: This has no effect, the guard goes out of scope on the next line so would be dropped here anyway.

File: src/winrtble/peripheral.rs [hunk 2]
```diff
@@ -143,26 +143,30 @@ impl Peripheral {
...
+                        (manufacturer_id, data)
+                    })
+                    .collect();
+
+                // Emit event of newly received advertisement
+                self.emit_event(CentralEvent::ManufacturerDataAdvertisement {
+                    id: self.shared.address.into(),
+                    manufacturer_data: manufacturer_data_guard.clone(),
+                });
+                drop(manufacturer_data_guard);
```
Comment (qwandor) [inline src/winrtble/peripheral.rs:168]: Ditto, the guard is about to go out of scope so this has no effect.

File: src/winrtble/peripheral.rs [hunk 3]
```diff
@@ -217,6 +221,7 @@ impl Peripheral {
                     id: self.shared.address.into(),
                     service_data: service_data_guard.clone(),
                 });
+                drop(service_data_guard);
```
Comment (qwandor) [inline src/winrtble/peripheral.rs:224]: Ditto.

File: src/winrtble/peripheral.rs [hunk 4]
```diff
@@ -253,6 +258,7 @@ impl Peripheral {
                     id: self.shared.address.into(),
                     services: services_guard.iter().copied().collect(),
                 });
+                drop(services_guard);
```
Comment (qwandor) [inline src/winrtble/peripheral.rs:261]: Ditto.

File: src/winrtble/peripheral.rs [hunk 5]
```diff
@@ -263,6 +269,7 @@ impl Peripheral {
                 BluetoothAddressType::Random => Some(AddressType::Random),
                 _ => None,
             };
+            drop(address_type_guard);
```
Comment (qwandor) [inline src/winrtble/peripheral.rs:272]: Again, this is just about to go out of scope and be dropped anyway.

Comment (jcaude) [reply src/winrtble/peripheral.rs:272]: Ok, I was not sure about that.. I discard all of them.

Comment (jcaude) [reply src/winrtble/peripheral.rs]: I just commit the update

**Comment by qdot** [comment]
Commit lined cleaned up and merged as [53f5d41](https://github.com/deviceplug/btleplug/commit/53f5d4134fafdf56aeeb53800c47709cdb012a8c)

**Pull request closed without merging**